src/DynamoPackages/PackageManagerClient.cs

```diff
@@ -77,11 +77,31 @@ internal IEnumerable<PackageHeader> ListAll()
         {
             return FailFunc.TryExecute(() => {
                 var nv = HeaderCollectionDownload.ByEngine("dynamo");
-                var pkgResponse = this.client.ExecuteAndDeserializeWithContent<List<PackageHeader>>(nv);
+                var pkgResponse = this.client.ExecuteAndDeserializeWithContent<List<PackageHeader>>(nv);                
+                CleanPackagesWithWrongVersions(pkgResponse.content);
                 return pkgResponse.content;
             }, new List<PackageHeader>());
         }
 
+        /// <summary>
+        /// For every package Checks its versions and exclude the ones with errors
+        /// </summary>
+        /// <param name="packages"></param>
+        void CleanPackagesWithWrongVersions(List<PackageHeader> packages)
+        {
+            foreach (var package in packages)
+            {
+                bool packageHasWrongVersion = package.versions.Count(v => v.full_dependency_versions == null) > 0;
```

Comment:
@zeusongit Can you remind me what does `full_dependency_versions` mean here?

Reply:
it contains all version of packages listed in `full_dependency_ids` these are packages on which the current package is dependent upon.

Reply:
I talked to greg marr about this. It was probably a DB error where a partial duplication occurred somehow.

Reply:
@zeusongit So this change is expected? Do we need to apply this elsewhere?

Reply:
There are 70 packages under the same scenario, I have added more details here https://jira.autodesk.com/browse/DYN-4977  , I'm checking in-depth in order to find the root cause

Reply:
Is the packages list here, a list of dependent packages?

Reply:
> Is the packages list here, a list of dependent packages?

not, they are packages that also have wrong data in their versions.